Makefile

```diff
@@ -116,10 +116,16 @@ cobertura:
 		grep -v zz_generated.deepcopy | \
 		$(GOCOVER_COBERTURA) > $(GO_TEST_OUTPUT)/cobertura-coverage.xml
 
-e2e-tag-images:
+e2e.test.images:
+	@$(INFO) Building E2E test images
+	@docker build --load -t $(BUILD_REGISTRY)/fn-labelizer-$(TARGETARCH) test/e2e/testdata/images/labelizer
```

Comment:
We can push this somewhere and just take it from there as we are doing for packages.

Reply:
These images are used only for running e2e tests, they do not have any other purpose, or extra value. IMHO, I do not see much value in keeping it in an external registry, please correct me if you see it differently. At least I would start like this, and if we see the value later, we can always then decide to push it somewhere.

Reply:
I was just suggesting to start the other way around as that looks easier to me and we are already doing that for other tests. In a separate issue we could agree on the best approach and handle them all the same way instead of building this one here, while having the packages built and pushed beforehand, see the image used here: https://github.com/crossplane/crossplane/blob/master/test/e2e/manifests/pkg/configuration/dependency/configuration.yaml

Reply:
True, however, provider-nop is used for other things like demo - this particular image was crafted sollely for purpose of asserting that pulling from registry works.. and the function just add a label to each MR. IMHO, I do not see a value of publishing such image to a registry, it should be more generic and configurable for that.

Reply:
> https://github.com/crossplane/crossplane/blob/master/test/e2e/manifests/pkg/configuration/dependency/configuration.yaml

@phisco  where does the source repo live for e2e-depends-on-provider-nop exist?

Reply:
`crossplane/e2e-depends-on-provider-nop:v0.1.0` doesn't look like a provider we are expecting other people to use...

Reply:
and for me it would be a reason more not to publish it to a public registry as Docker hub. Pulling it from there could be sometimes problematic, given the known Dockerhub rate limiting. I would propose that we avoid publishing the image prematurely now in this PR - let it bake for some time, and if we see the value of consuming it in a context different from e2e testing, we could move it in a separate repo, and introduce workflows and publishing logic. If we wanna iterate fast, keeping the image here is beneficial for now.

Reply:
Yes, but I'd rather have a single approach for now, so given that we are already building and pushing the other image separately, I'd keep it simple and use the same approach here, postponing a better decision to another issue.

Reply:
> Yes, but I'd rather have a single approach for now, so given that we are already building and pushing the other image separately, I'd keep it simple and use the same approach here, postponing a better decision to another issue.

Moving test images in a separate repo is going to establish a separate lifecycle for them. Everytime we need to tweak them for some tests, we are going to have two PRs instead of one and it might happen that we need to merge first image PR, and then to realize that the change was not right for e2e tests, so we need to do another rounds of PR. Here, everything is kept within single PR. @negz would like to hear your 2 cents on this subject?

Reply:
Just to be clear, I'm not suggesting moving them to a separate repo, but more like, stick to the status quo and open another issue to solve it organically for all images used by e2es. E.g. we could think of having a dedicated directory `test/e2e/images` with subfolders for all images uses and build and push them all through a separate pipeline or step in the ci pipeline

Reply:
I do understand your intention @phisco but given that we just started adding e2e tests, not sure that we have any status quo yet. It would be great if we could treat lifecycle of all used images in the same way, but I rather see it as a graduation process:

* an image gets introduced within this repo due to a necessity to assert some crossplane functionality. 
* IMHO, we should always try to use a public image as possible, but we are going to have often situation that none of the images is specific enough for the given usecase (e.g. if we need to test some error case, we should craft an image that produces such error)
* If we see that the image is actually useful outside of e2e testing context, then we can promote it and move it to its own repo and publish in a public registry.

> we could think of having a dedicated directory test/e2e/images with subfolders for all images uses and build and push them all through a separate pipeline or step in the ci pipeline

This PR introduces `test/e2e/testdata/images` dir.

Reply:
Fine, as you want, but please create an issue about properly handling images used by E2Es and a TODO here referencing it.